### DISCUSSION TITLE
Replace substring in package name/branchName

### DISCUSSION THREAD

**Discussion by jpgoelz** [renovatebot/renovate#26234] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.90.1

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### What is your question?

In our company, we are using a Docker proxy to reduce the amount of images downloaded from the Docker hub directly.

This means, all our images are declared using the Docker registry URL like, e.g.,
```dockerfile
FROM harbor.example.com/docker/library/nginx:1.25.3-alpine-slim
```
in Dockerfiles or
```yaml
image: harbor.example.com/docker/library/maven:3.9.5-eclipse-temurin-17
```
in our _.gitlab-ci.yml_.

This leads to Renovate adding the URL-prefix to the package name and thus to the title, and branch names as well:

![image](https://github.com/renovatebot/renovate/assets/32487587/6b7d7266-1d2a-4af2-bc78-b7a8eb3e2585)

Is there a way to remove the "harbor.example.com/docker/library/" part?

I tried to at least remove it from the `branchTopic` by trying to invoke the replace function in handlebars [as mentioned in the docs](https://docs.renovatebot.com/templates/#additional-handlebars-helpers) like as follows, because we have a character limit of 65 characters which is often exceeded.

```json5
{
  // Remove the Docker proxy url from the branch name.
  matchPackagePatterns: ["^harbor.example.com/docker/library/"],
  branchName: "{{{branchPrefix}}}{{{additionalBranchPrefix}}}{{{replace 'harbor\.example\.com-docker-library-' '' branchTopic}}}",
},
```

This did not work (Is there some kind of mistake, perhaps?).

**Comment by viceice** [comment]
try registry aliases 

https://docs.renovatebot.com/configuration-options/#registryaliases

**Reply by jpgoelz** [reply]
This did not work, unfortunately. But maybe I did it wrong.

Where/how do I configure this?

I placed it in my renovate.json5 like this:
```json5
{
  $schema: "https://docs.renovatebot.com/renovate-schema.json",
  extends: [
    /* ... */
  ],
  dockerfile: {
    registryAliases: {
      "docker.io": "harbor.example.com/docker",
    },
  },
  packageRules: [
    /* ... */
  ],
}
```

I also tried adding it to the config.js of our self-hosted Renovate .

**Reply by viceice** [reply]
registry aliases need to be at config root

**Reply by jpgoelz** [reply]
Ooooh, I see. 🤦 You mean, instead of 

```json5
  dockerfile: {
    registryAliases: {
      "docker.io": "harbor.example.com/docker",
    },
  },
```

use 

```json5
  registryAliases: {
    "docker.io": "harbor.example.com/docker",
  },
```
directly?

This has had no effect either. Again, I tried this in the project's renovate.json5 as well as the config.js of our self-hosted Renovate instance.

**Reply by viceice** [reply]
please create a minimal reproduction.

what are the logs saying about detected dependencies?

**Reply by jpgoelz** [reply, marked as answer]
I got it working now. It seems that I had to reverse the aliases from
`"docker.io": "harbor.example.com/docker"` to `"harbor.example.com/docker": "docker.io"`!

The Packages are now displayed as `docker.io/...`  While in the code, only the version is updated. 🎉 

<img src="https://github.com/renovatebot/renovate/assets/32487587/1100c5e4-bbd9-4cbc-8639-258c86978242" width="50%" />

<img src="https://github.com/renovatebot/renovate/assets/32487587/cf51866c-3b68-4d9f-b7d1-ddd7dc64cca2" width="50%" />

**Reply by jpgoelz** [reply]
This is fine so far, although I would prefer it if the repository was not listed at all (just `library/eclipse-temurin` instead of `docker.io/library/eclipse-temurin`). I tried `"harbor.example.com/docker": ""`, but it would then say 

WARN: Package lookup failures
```json
{
  "warnings": [
    "Failed to look up docker package harbor.example.com/docker/library/eclipse-temurin",
  ]
  "files": [
    ".gitlab-ci.yml"
  ]
}
```

**Reply by jpgoelz** [reply]
Oh, and thank you for your help, of course!
